redis-5.0.5/src/rbuf/test/rbuf_test.c

```diff
@@ -0,0 +1,112 @@
+/***********************************************************/
+/*********************RingBuf : Test*************************/
+/************Lock Free Data Structure***********************/
+/***********************************************************/
+
+/* A ring buffer for multi-threaded programs. A single producer pushes
+ * objects into the ring, while a single consumer pops them off for
+ * processing. If the ring is full (i.e., the consumer should be 
+*  slower than the producer), then the push operation fails.
+*/
+
+
+
+/* 
+ * a test consol to see if ringbuffer's exceptions are working fine
+ * with exceeding push and pop options
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "../ringbuf.h"
+
+#define     MAXLEN      100
+
+
+
+int main() {
+    
+	const int buffer_size = 6;
+	const int start_location = 0;
+
+	int push_size_1 = 8;
+	int get_size_1 = 2;
+	// int push_size_2 = 6;
+	int get_size_2 = 12;
+
+	void* data[buffer_size];
+	void* get_data;
+
+	struct ring* ring_buffer = init_ring( buffer_size );
+	struct consumer *con_1 = init_consumer( start_location );
+	struct consumer *con_2 = init_consumer( start_location );
+
+	int i;
+
+	for( i = 0; i < push_size_1 ; i++){
+
+		data[i] = malloc(sizeof(int));
```

Comment:
int *tmp_int = (int *) malloc(sizeof int);
assert(tmp_int == NULL);
*tmp_int = i;

Reply:
Thanks , will feed in Real Data later